patchwork-extensions-block/src/main/java/net/patchworkmc/mixin/extensions/block/MixinWorld.java

```diff
@@ -34,7 +37,20 @@
 public abstract class MixinWorld {
 	// if (blockState.getBlock().hasBlockEntity()) {
 	@Redirect(method = "breakBlock", at = @At(value = "INVOKE", target = Signatures.BlockState_getBlock, ordinal = 0))
-	public Block patchwork_breakBlock_getBlock(BlockState blockstate) {
+	private Block patchwork_breakBlock_getBlock(BlockState blockstate) {
 		return BlockContext.hasBlockEntityBlockMarker(blockstate);
 	}
+
+	////////////////////////////////////////////////////////
+	/// doesAreaContainFireSource - IForgeBlock.isBurning
+	////////////////////////////////////////////////////////
+	// This really should be included in the Fabric API!
+	// Block block = this.getBlockState(pooledMutable.set(o, p, q)).getBlock();
+	// if (block == Blocks.FIRE || block == Blocks.LAVA) {
+	@Redirect(method = "doesAreaContainFireSource", at = @At(value = "INVOKE", target = Signatures.World_getBlockState, ordinal = 0))
+	private BlockState patchwork_doesAreaContainFireSource_getBlockState(World world, BlockPos blockPos) {
+		BlockState blockState = world.getBlockState(blockPos);
+		boolean isBurning = ((IForgeBlockState) blockState).isBurning(world, blockPos);
+		return isBurning ? Blocks.FIRE.getDefaultState() : Blocks.WATER.getDefaultState();
```

Comment:
I'd return the actual blockstate instead of `Blocks.WATER.getDefaultState()` to mimic vanilla behavior better for a potential Fabric mod

Reply:
This the return value is used to by pass the vanilla code `if (block == Blocks.FIRE || block == Blocks.LAVA`.

In 1.16, They moved World.doesAreaContainFireSource to Entity.method_29496(BlockState) and EntityType.method_27138(BlockState). But the addition of soul fire and campfire make the case much more complicated. I'd like to see how forge reacts to this change.